code/src/reducers/tasks.js

```diff
@@ -0,0 +1,49 @@
+import {createSlice} from '@reduxjs/toolkit'
+
+const taskList = [
+    {id: 1, text: 'My first todo', complete: true, date: new Date().getTime(), due: null},
+    {id: 2, text: 'My second todo', complete: false, date: new Date().getTime(), due: null},
+    {id: 3, text: 'My third todo', complete: true, date: new Date().getTime(), due: null},
+    {id: 4, text: 'My fourth todo', complete: false, date: new Date().getTime(), due: null}
+]
+
+
+
+
+export const tasks = createSlice({
+    name: 'tasks',
+    initialState: {
+        items: taskList},
+    reducers: {
+        addTask: (state, action) => {
+            const latestTaskId = state.items.map(task => task.id)
+            const {startDate, newTask} = action.payload
+            state.items.push({
+                id: (state.items.length > 0 ? Math.max(...latestTaskId) +1 : 1), 
+                text: newTask, 
+                complete: false,
+                date: new Date().getTime(),
+                due: startDate
+            });
+        },
+
+        clearTasks: (state) => {
+            state.items.length = 0;
+        },
```

Comment:
Interesting! If you'd have an empty initial state you could return that (or just return an empty array)